### PR TITLE
add docker files for integration

### PR DESCRIPTION
**Pull request by siepet**
Add docker compose yaml file, docker deploy integration and Dockerfile for integration ( dynamic stagings ).

### CODE REVIEW

File: config/deploy/integration.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+server 'people.integration.devguru.co', user: 'people', roles: %w(app db web)
+
+set :branch, ENV["REVISION"] || "master"
+
+set :deploy_tom "/home/deploy/apps/#{fetch(:application)}/integration"
```
Comment (netguru-hound) [inline config/deploy/integration.rb]: unexpected token tSTRING_BEG
(Using Ruby 2.3 parser; configure using `TargetRubyVersion` parameter, under `AllCops`)

File: config/deploy/integration.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+server 'people.integration.devguru.co', user: 'people', roles: %w(app db web)
+
+set :branch, ENV["REVISION"] || "master"
+
+set :deploy_to, "/home/deploy/apps/#{fetch(:application)}/integration"
+set :pull_request, -> { ENV["PULL_REQUEST_NUMBER"] }
+
+set :docker_compose, -> { true }
+set :docker_compose_project_name, -> { "#{fetch(:application)}#{fetch(:pull_request)}" }
+set :docker_pass_env, -> { ["PULL_REQUEST_NUMBER"] }
```
Comment (netguru-hound) [inline config/deploy/integration.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: config/deploy/integration.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+server 'people.integration.devguru.co', user: 'people', roles: %w(app db web)
+
+set :branch, ENV["REVISION"] || "master"
+
+set :deploy_to, "/home/deploy/apps/#{fetch(:application)}/integration"
+set :pull_request, -> { ENV["PULL_REQUEST_NUMBER"] }
```
Comment (netguru-hound) [inline config/deploy/integration.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: config/deploy/integration.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+server 'people.integration.devguru.co', user: 'people', roles: %w(app db web)
+
+set :branch, ENV["REVISION"] || "master"
```
Comment (netguru-hound) [inline config/deploy/integration.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: docker/integration/entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+set -e
+
+function start {
+  # executed when container is run
+  bundle exec whenever --update-crontab --set environment=staging
```
Comment (rindek) [inline docker/integration/entrypoint.sh]: There should be tasks for setup database  and precompile assets

**Comment by lubieniebieski** [comment]
https://github.com/netguru/people/blob/master/.rubocop.yml#L36

File: docker/integration/entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+set -e
+
+function start {
+  # executed when container is run
+  rake assets:precompile
+  rake db:setup
+  rake db:seed
```
Comment (rindek) [inline docker/integration/entrypoint.sh]: db:setup is enough, it will do db:seeds

File: docker/integration/entrypoint.sh [hunk 7]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+set -e
+
+function start {
+  # executed when container is run
+  rake assets:precompile
```
Comment (rindek) [inline docker/integration/entrypoint.sh]: put this after db is prepared - rails env likes to scream b/c of that

**Pull request merged**